src/device/mod.rs

```diff
@@ -336,13 +339,24 @@ struct Runtime {
 }
 
 impl Device {
+    fn check_ipv6_availability() -> bool {
+        if_addrs::get_if_addrs()
+            .unwrap_or(vec![])
+            .into_iter()
+            .any(|interface| interface.ip().is_ipv6())
+    }
+
     // Incoming changes should fix
     #[allow(unwrap_check)]
     pub fn new<F: EventCb>(
-        features: Features,
+        mut features: Features,
         event_cb: F,
         protect: Option<Protect>,
     ) -> Result<Self> {
+        if features.ipv6 {
```

Comment:
There is no need to base this on the actual availability (fallback mechanisms are designed for that). Therefore it is enough to base it on `self.features.ipv6` instead of checking interface availability.

Also it is extremely likely, that when calling `device::new` the IP will not be set into the interface itself.